draft-ietf-ohai-ohttp.md

```diff
@@ -1258,21 +1258,25 @@ interception might choose to disable Oblivious HTTP in order to ensure that
 content is accessible to middleboxes.
 
 
-# Repurposing the Encapsulation Format
+# Repurposing the Encapsulation Format {#repurposing}
 
 The encrypted payload of an OHTTP request and response is a binary HTTP
 message {{BINARY}}. Client and target agree on this encrypted payload type by
 specifying the media type "message/bhttp" in the HPKE info string and HPKE
 export context string for request and response encryption, respectively.
 
 Future specifications may repurpose the encapsulation mechanism described in
-{{hpke-encapsulation}}, provided that the content type of the encrypted
-payload is appropriately reflected in the HPKE info and context strings. For
-example, if a future specification were to use the encryption mechanism in
-this specification for DNS messages, identified by the "application/dns-message"
-media type, then the HPKE info string SHOULD be "application/dns-message
-request" for request encryption, and the HPKE export context string should be
-"application/dns-message response" for response encryption.
+{{hpke-encapsulation}}.  This requires that the specification define a new media
+type.  The encapsulation process for that content type can follow the same
```

Comment:
Well, the specification doesn't need a _new_ media type. It can certainly use other or existing media types. It just needs to be tied to a particular media type.

Reply:
It really does.  Unless it decides that it can use some other way to identify how the payload of the message should be interpreted.  In the OHTTP case - and likely in others - you can use ambient information, like the fact that the message appears in an HTTP request and for a particular resource, but that isn't particular portable, nor does it permit changing the format of content.